agent/spec/spec_helper.rb

```diff
@@ -74,6 +74,11 @@
     end
   end
 
+  config.around :each do |ex|
+    Timeout.timeout(5.0) do
```

Comment:
The EM specs need to wait for some expected method to get called before they can `EM.stop`. If that method doesn't get called, then the specs just time out. Typically that happens because when some spec double expectation fails, the `WebsocketClient` rescues and logs it as an error, without propagating it out and letting `EM.run { ... }` fail.